service/discovery/extra/csaf/provider_discover.go

```diff
@@ -69,7 +69,10 @@ func (d *csafDiscovery) handleProvider(lpmd *csaf.LoadedProviderMetadata) (resou
 		Raw: discovery.Raw(pmd),
 	}
 
-	keys := d.discoverKeys(pmd.PGPKeys)
+	// TODO(oxisto): find a sensible ID instead of this one
```

Comment:
Missing tests for functions: discoverProviders, handleProvider,

Reply:
This is implicitly tested when we test the list, so I would skip it here

Reply:
I think it's fine for now, but in general we should write a test for every function/metho, whether it's already implicitly tested or not.